src/avk.cpp

```diff
@@ -2314,17 +2314,29 @@ namespace avk
 		
 		return result;
 	}
-	
+
 	std::optional<command_buffer> buffer_t::fill(const void* aDataPtr, size_t aMetaDataIndex, sync aSyncHandler)
 	{
 		auto metaData = meta_at_index<buffer_meta>(aMetaDataIndex);
 		auto bufferSize = static_cast<vk::DeviceSize>(metaData.total_size());
+		return fill_partially(aDataPtr, bufferSize, 0u, aMetaDataIndex, std::move(aSyncHandler));
+	}
+
+	std::optional<command_buffer> buffer_t::fill_partially(const void* aDataPtr, size_t aDataSizeInBytes, size_t aOffset, size_t aMetaDataIndex, sync aSyncHandler)
+	{
+		auto dataSize = static_cast<vk::DeviceSize>(aDataSizeInBytes);
 		auto memProps = memory_properties();
 
+#ifdef _DEBUG
+		auto metaData = meta_at_index<buffer_meta>(aMetaDataIndex);
+		if (aOffset + aDataSizeInBytes > metaData.total_size())
+			throw avk::runtime_error("The fill operation would write beyond the buffer's size.");

```

Comment:
```suggestion
		if (aOffset + aDataSizeInBytes > metaData.total_size()) {
			throw avk::runtime_error("The fill operation would write beyond the buffer's size.");
		}

```

That's great! Please just add the curly brackets around code blocks according to coding conventions!